### ISSUE TITLE
In Scala mode unable to insert spaces between words using SPC

### ISSUE THREAD

**Issue by randre03** [syl20bnr/spacemacs#6059]
#### Description
I just recently installed and began using Scala Mode. I have noticed that as soon as I open a new file with the type *.scala I cannot include a space in between the first and second word using the spacebar. Sometimes a combination of using the Ctrl and/or Alt keys and then pressing SPC will work. Sometimes not.

#### Reproduction guide
- Start Emacs
- SPC F F
- Create a new .scala file
- Once file opens, try to type Object Foo as if you were creating a new scala object named Foo
- Try to place a space in between the words "Object" and "Foo" using SPC bar
- If not possible, try typing C+M+SPC, or C+SPC or M+SPC

*Observed behaviour:*
If the issue is reproduced then the tester will see ObjectFoo as opposed to Object Foo and will not be able to separate the two words by using the SPC Bar. The tester might be able to insert a space after trying one of the other alternative, undesirable key sequences listed above.

*Expected behaviour:*
Expected (meaning desired) behavior would be to be able to type Object Foo by simply pressing the SPC bar in between the words "Object" and "Foo".

#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.20
- Spacemacs branch: master (rev. 9c7250a)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
((auto-completion :variables auto-completion-return-key-behavior 'complete auto-completion-tab-key-behavior 'complete auto-completion-complete-with-key-sequence nil auto-completion-enable-sort-by-usage t auto-completion-enable-help-tooltip t auto-completion-enable-snippets-in-popup t)
 better-defaults clojure elixir elm emacs-lisp git github html
 (ibuffer :variables ibuffer-group-buffers-by 'modes)
 javascript markdown ocaml org osx racket react rust scala semantic
 (shell :variables shell-default-shell 'ansi-term shell-default-height 25 shell-default-position 'bottom)
 spell-checking syntax-checking version-control)
```

#### Backtrace
```
Not provided.

**Comment by StreakyCobra** [comment]
I can't reproduce this on `develop` with packages updated. @randre03 Can you try the `develop` branch and be sure that your packages are up to date?

**Comment by randre03** [comment]
I just updated to `develop` and updated all packages and the same unwanted behavior occurred. Here are the updated Spacemacs details:

#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.20
- Spacemacs branch: develop (rev. 1f993bd)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
((auto-completion :variables auto-completion-return-key-behavior 'complete auto-completion-tab-key-behavior 'complete auto-completion-complete-with-key-sequence nil auto-completion-enable-sort-by-usage t auto-completion-enable-help-tooltip t auto-completion-enable-snippets-in-popup t)
 better-defaults clojure elixir elm emacs-lisp git github html
 (ibuffer :variables ibuffer-group-buffers-by 'modes)
 javascript markdown ocaml org osx racket react rust scala semantic
 (shell :variables shell-default-shell 'ansi-term shell-default-height 25 shell-default-position 'bottom)
 spell-checking syntax-checking version-control)

**Comment by StreakyCobra** [comment]
Ok. For me the following steps don't exhibit this behaviour:

- Start emacs
- `SPC f J scala RET`
- `i Object Foo ESC`

Maybe it's a platform problem, or because I don't know scala and do something wrong… Anyway let's see if someone else encounter it.

**Comment by randre03** [comment]
I think I have figured out the issue that was causing this. My setup had Aggressive Indent Mode turned on at all times. Once Aggressive Indent Mode was turned off I am able to type in .scala files with no trouble. 

Based on the instructions at https://github.com/Malabarba/aggressive-indent-mode I added `'scala-mode` to `'aggressive-indent-excluded-modes` and now things are working without trouble.

In my mind, this can be closed now.
